### PR TITLE
[NEW] Timesheet Creation Tool

### PR DESCRIPTION
**Pull request by shreyashah115**
#9434 
Summary:
1. New DocType Timesheet Creation Tool.
2. Add multiple employees and create their individual timesheets based on same `time_logs`.
3. Buttons `create_timesheet_as_draft` and `create_and_submit_timesheet`.
4. Moved the common functionalities of Timesheet and Timesheet Creation Tool DocTypes to `timesheet_common.js` file.
5. Added a test case.

timesheetfinal

### CODE REVIEW

File: erpnext/projects/doctype/timesheet/timesheet.js [hunk 1]
```diff
@@ -1,36 +1,9 @@
...
-
-		frm.fields_dict['time_logs'].grid.get_field('project').get_query = function() {
-			return{
-				filters: {
-					'company': frm.doc.company
-				}
-			}
-		}
-	},
+{% include 'erpnext/projects/timesheet_common.js' %};
```
Comment (tundebabzy) [inline erpnext/projects/doctype/timesheet/timesheet.js]: A couple of things I noticed:
screenshot-2018-1-10 timesheet creation tool 1

After checking 'Bill', The fields with red arrows are not auto filled as usual. Check Timesheet form for the normal behavior.

screenshot-2018-1-10 timesheet creation tool 2
The 'View Timesheets' link is not obvious to the user as a link.

**Review by tundebabzy — CHANGES_REQUESTED** [review]

File: erpnext/projects/doctype/timesheet/timesheet.js (hunk 1 above)
Comment (shreyashah115) [reply erpnext/projects/doctype/timesheet/timesheet.js]: Hi @tundebabzy, `billing_hours` get updated only if `Maintain Billing Hours and Working Hours Same on Timesheet` field is checked in HR Settings. Also, the other fields such as `billing_rate` and `costing_rate` get updated if the respective rates are found in Activity Type DocType of that particular Activity.

**Review by tundebabzy — DISMISSED** [review]
Create and Submit Is not working properly
peek 2018-01-11 11-12

**Review by tundebabzy — APPROVED** [review]

**Comment by dufani1** [comment]
@tundebabzy is the time sheets work for calculating project costs even the bill check box not checked? I think should work.

**Comment by umairsy** [comment]
How will the scenario like over-time calculation get's addresses in it? If Empployees have worked for overtime for different From and To Time? Will it fetch From Time and To Time for the specific Employee in the Salary Slip?

**Comment by shreyashah115** [comment]
@umairsy It only helps in creating bulk timesheets for employees, instead of duplicating timesheets and submitting for the same activities and time, this single DocType could be used. The other functionalities like creating Salary slips work from the submitted Timesheet docs itself.

**Comment by albertusgeyser** [comment]
Could you implement as mentioned in issue (Feature Request):
#12541

**Comment by rohitwaghchaure** [comment]
- [ ] Convert single page to transactions doctype same as Payroll Entry, it will help to keep track.

- [ ]  Add posting date in the doctype

- [ ] Add checkbox Same Time Slots for All, if it's enabled then use same time slots for all employee while making timesheet else add provision to select employee against the time slots in the child table and group the time slots by the employees

- [ ] Add provision to make sales invoice from the timesheet tool

Closing PR, please make above changes and reopen it

**Comment by albertusgeyser** [comment]
Features added or not yet? #12541

**Comment by shreyashah115** [comment]
Hi @albertusgeyser 
If you're in such a hurry, why not create a [bounty](https://erpnext.org/new-bounty) for the same?

**Comment by mhbu50** [comment]
@shreyashah115 how much we have to put to get this?

**Comment by mhbu50** [comment]
@shreyashah115 any update on my request?

**Pull request closed without merging**